private/pkg/storage/diff.go

```diff
@@ -238,16 +263,16 @@ func Diff(
 				diffOptions.toDiffPackageOptions()...,
 			)
 			if err != nil {
-				return err
+				return nil, err
 			}
 			if len(diffData) > 0 {
 				if _, err := writer.Write(diffData); err != nil {
-					return err
+					return nil, err
 				}
 			}
 		}
 	}
-	return nil
+	return changedPaths, nil
```

Comment:
Can you add a comment that this is sorted? And document that the return value is sorted?

Reply:
It is not currently sorted. Do you think it should be? The way it is currently used, it does not need to be.

It isn't necessarily sorted because it is constructed in two passes. While each pass is over a sorted slice, the second pass could append items that should be inserted if we need it to be sorted.

Reply:
I think it should be, to make us really explicit what is being returned. Simple to do anyways

Reply:
Okay. Now always sorted and documented, too.